lib/swaps/Swaps.ts

```diff
@@ -336,7 +341,7 @@ class Swaps extends EventEmitter {
     if (errMsg) {
       this.logger.error(errMsg);
       this.failDeal(deal, SwapFailureReason.SwapClientNotSetup, errMsg);
-      return;
+      return SwapFailureReason.SwapClientNotSetup;
```

Comment:
Lines 496, 500, 550: all these swap failures won't be returned to the linked `executeSwap`. To achieve this more work is necessary to link between the async events.
Same for more failures in `Swap.js`, which some of them can be linked to the taker swap call.

Reply:
I'm not sure I follow, in these cases, calling `failDeal` will in turn set the failure reason on the deal and emit the `swap.failed` event. `executeSwap` is listening for that event and will return the failure reason if it occurs. Am I missing something that would cause this not to work?

Reply:
Indeed, I got confused and didn't notice that the events are already connected. Forget about it!